app/src/main/java/com/example/trainingintive/repository/DogImageRepository.kt

```diff
@@ -1,11 +1,19 @@
 package com.example.trainingintive.repository
 
 import com.example.trainingintive.model.DogImageUrl
+import com.example.trainingintive.repository.local.DogImageDao
 import com.example.trainingintive.repository.network.DogImageApiService
 import io.reactivex.rxjava3.core.Single
 import javax.inject.Inject
 
-class DogImageRepository @Inject constructor(private val dogImageApiService: DogImageApiService) {
+class DogImageRepository @Inject constructor(
+    private val dogImageApiService: DogImageApiService,
+    private val dogImageDao: DogImageDao
+) {
 
     fun getDogImageUrl(): Single<DogImageUrl> = dogImageApiService.getDogImage()
+
+    fun getAllDogImagesFromLocalDatabase(): List<DogImageUrl> = dogImageDao.getAllDogImageUrls()
+
+    fun insertDogImageIntoLocalDatabase(dogImageUrl: DogImageUrl) = dogImageDao.insert(dogImageUrl)
```

Comment:
Minor: This name can be also simplified to something like `insertIntoLocalDatabase` because we are in the scope of `DogImageRepository` and we have only one insert, so it's logical that we are inserting DogImage.